classes/app.rb

```diff
@@ -2,7 +2,25 @@
 require_relative './teacher'
 require_relative './book'
 require_relative './rental'
-require_relative './main'
+
+def perform_tasks(input)
+  case input
+  when 1
+    list_all_books
+  when 2
+    list_all_people
+  when 3
+    create_a_person
+  when 4
+    create_a_book
+  when 5
+    create_a_rental
+  when 6
+    show_rentals_for_person
+  else
+    puts 'Invalid option. Retry'
+  end
+end
```

Comment:
- [ ] according to the requirements all requests for input and input validation should've been placed in the main.rb file and/or classes/methods whose responsibilities are to request or validate input. right now you're printing to the console, getting en processing data al in the same methods. please have a look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) for more details

https://github.com/sameedmohsin28/OOP-school-library/blob/3f6adbf614d17a09b326979bbbc0733fb3bcfe1f/classes/app.rb#L30-L145

Reply:
@mikethreels 

Could you please further elaborate your point? I can not exactly understand the changes you are requesting.
Thanks

Reply:
Hi @sameedmohsin28, this is part of the solid principle. If you need extra guidance it might be a good idea to request for a one-on-one session with a senior CR.